e2e/tests/core/02-client/client_test.go

```diff
@@ -153,6 +154,55 @@ func (s *ClientTestSuite) TestClientUpdateProposal_Succeeds() {
 	})
 }
 
+// TestScheduleIBCUpgrade_Succeeds tests that a governance proposal to schedule an IBC software upgrade is successful.
+func (s *ClientTestSuite) TestScheduleIBCUpgrade_Succeeds() {
+	t := s.T()
+	ctx := context.TODO()
+
+	_, _ = s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+
+	s.Require().NoError(test.WaitForBlocks(ctx, 10, chainA, chainB))
+
+	authority, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+	s.Require().NoError(err)
+	s.Require().NotNil(authority)
+
+	var (
+		originalTrustingPeriod time.Duration
+		newTrustingPeriod      time.Duration
+	)
+
+	t.Run("send schedule IBC upgrade message", func(t *testing.T) {
+		authority, err := s.QueryModuleAccountAddress(ctx, govtypes.ModuleName, chainA)
+		s.Require().NoError(err)
+		s.Require().NotNil(authority)
+
+		clientState, err := s.QueryClientState(ctx, chainA, ibctesting.FirstClientID)
+		s.Require().NoError(err)
+
+		originalTrustingPeriod = clientState.(*ibctm.ClientState).TrustingPeriod
+		newTrustingPeriod = originalTrustingPeriod + time.Duration(time.Hour*24)
+
+		upgradedClientState := clientState.(*ibctm.ClientState)
+		upgradedClientState.TrustingPeriod = newTrustingPeriod
+
+		latestHeight := clientState.GetLatestHeight().GetRevisionHeight()
```

Comment:
might be nicer to reference the actual chain height, can we obtain it through chainA?